src/Parser/Monk/Windwalker/Modules/Spells/SpinningCraneKick.js

```diff
@@ -0,0 +1,178 @@
+      return;
+    }
+    // The event doesn't specify instance on the first target of that type
+    if (targetInstance === undefined) {
+      targetInstance = 1;
+    }
+    const markoftheCraneTarget = {targetID: event.targetID, targetInstance: targetInstance, timestamp: event.timestamp };
+    this.markoftheCraneTargets.push(markoftheCraneTarget);
+  }
+
+  on_byPlayer_refreshdebuff(event) {
+    const spellId = event.ability.guid;
+    let targetInstance = event.targetInstance;
+    if (spellId !== SPELLS.MARK_OF_THE_CRANE.id) {
+      return;
+    }
+    if (targetInstance === undefined) {
+      targetInstance = 1;
+    }
+    const markoftheCraneTarget = {targetID: event.targetID, targetInstance: targetInstance, timestamp: event.timestamp };
+    let i = 0;
+    while (i <= this.markoftheCraneTargets.length - 1) {
+      if (this.markoftheCraneTargets[i].targetID === markoftheCraneTarget.targetID && this.markoftheCraneTargets[i].targetInstance === markoftheCraneTarget.targetInstance) {
+        this.markoftheCraneTargets[i].timestamp = markoftheCraneTarget.timestamp;
+      }
+      i++;
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SPINNING_CRANE_KICK.id) {
+      return;
+    }
+    this.markoftheCraneStacks = 0;
+    let i = 0;
+    while (i < this.markoftheCraneTargets.length) {
+      // removing expired targets to avoid looking through huge arrays in logs with a lot of targets
+      if (event.timestamp - this.markoftheCraneTargets[i].timestamp > 15000) {
+        this.markoftheCraneTargets.splice(i, 1);
+      }
+      else {
+        this.markoftheCraneStacks++;
+      }
+      i++;
+    }
+    // Currently only marking casts with lower DPET than Blackout Kick
+    // TODO: Expand to also mark targets with lower DPChi than Blackout Kick
+    if (this.markoftheCraneStacks <= 1) {
+      this.badCasts += 1;
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid; 
+    if (spellId !== SPELLS.SPINNING_CRANE_KICK_DAMAGE.id) {
+      return;
+    }
+    this.spinningCraneKickDuration = 1500 / (1 + this.statTracker.hastePercentage(this.statTracker.currentHasteRating));
+    this.spinningCraneKickHits++;
+    console.log("mark of the crane stacks", this.markoftheCraneStacks, event.timestamp);
+    // Spinning Crane Kick deals damage 4 times over 1.5 seconds (reduced by haste)
+    // This makes sure it only counts once per cast and only on casts that hit something
+    if (event.timestamp - this.lastSpinningCraneKickTick > this.spinningCraneKickDuration) {
+      this.totalMarksDuringHits += this.markoftheCraneStacks;
+      this.lastSpinningCraneKickTick = event.timestamp;
+    }
+  }
+
+  suggestions(when) {
+    when(this.badCasts).isGreaterThan(0).addSuggestion(
+      (suggest, actual, recommended) => {
+        return suggest(<span>You have ineffecient casts of <SpellLink id={SPELLS.SPINNING_CRANE_KICK.id} /> </span>)
```

Comment:
Please use the `Wrapper` component from `common/Wrapper` instead of `span` when the container has no function.